t/14-grutasks.t

```diff
@@ -608,6 +609,40 @@ subtest 'download assets with correct permissions' => sub {
     ok -f $assetpath, 'asset downloaded';
 };
 
+subtest 'finalize job results' => sub {
+    my %settings = (
+        DISTRI  => 'Unicorn',
+        FLAVOR  => 'pink',
+        VERSION => '42',
+        BUILD   => '666',
+        ISO     => 'whatever.iso',
+        MACHINE => "RainbowPC",
+        ARCH    => 'x86_64',
+        TEST    => 'minion',
+    );
+
+    my $job = $schema->resultset('Jobs')->create_from_settings(\%settings);
+    $job->discard_changes;
+    $job->insert_module({name => 'a', category => 'a', script => 'a', flags => {}});
+    $job->update_module('a', {result => 'ok', details => [{title => 'wait_serial', text => 'a-0.txt'}]});
+    $job->insert_module({name => 'b', category => 'b', script => 'b', flags => {}});
+    $job->update_module('b', {result => 'fail', details => [{title => 'wait_serial', text => 'b-0.txt'}]});
+    $job->update;
+    $job->discard_changes;
+    path($job->result_dir, 'a-0.txt')->spurt('Foo');
+    path('t/data/14-module-b.txt')->copy_to($job->result_dir . '/b-0.txt');
```

Comment:
You can't just use `spurt` here? I don't care much but it would likely be more elegant than needing an extra file. Since `spurt` accepts bytes you likely need `use Encode qw(encode); … encode('UTF-8', '正解')`.

Reply:
I spent an hour yesterday trying to get the expected Unicode string right *in Perl*. So I'm keeping the extra file because it gives me certainty that `b-0.txt` will really contain `正解` instead of some binary garbage like `æ­£è§£`. That's what I was actually writing to the file in the first iteration of the test.

Reply:
Ok. Certainty what the code does for non-Perl experts is actually a good argument :-)